styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java

```diff
@@ -263,6 +266,11 @@ private boolean shouldExecute(RunState runState) {
   }
 
   private void sendDequeue(InstanceState instanceState) {
+    if (!dequeueRateLimiter.tryAcquire()) {
+      LOG.debug("Dequeue rate limited");
+      return;
```

Comment:
I removed the code path fork, promoted `limitAndDequeue` to be an actual method and merged `evaluateResourcesForDequeue` into it. Now `tick` calls `limitAndDequeue` on each eligible instance until it returns false to signal that no further dequeue is currently possible.

wdyt?